tests/test_redis_lock.py

```diff
@@ -499,3 +500,15 @@ def workerfn(unblocked):
     worker.join(0.5)
 
     assert unblocked.value == 1
+
+
+def test_auto_renewal_stops_after_gc(conn):
+    """Auto renewal stops after lock is garbage collected."""
+    lock = Lock(conn, 'spam', auto_renewal=True, expire=1)
+    name = lock._name
+    lock.acquire()
+    del lock
+    gc.collect()
+
+    time.sleep(2)
+    assert conn.get(name) is None
```

Comment:
@ionelmc It fails on PyPy, because it's garbage collection works in somewhat different way and I have no idea, how to fix it.